flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java

```diff
@@ -301,6 +302,8 @@ private void cleanupExpiredVersionInState(long currentWatermark, List<RowData> r
     public void cleanupState(long time) {
         leftState.clear();
         rightState.clear();
+        nextLeftIndex.clear();
+        registeredTimer.clear();
```

Comment:
Thanks for the contribution @zicat , but I didn't understand why ` nextLeftIndex` and `registeredTimer`  state will lead to the data expansion, could you explain more and also add test in `TemporalRowTimeJoinOperatorTest` ?

Reply:
In our company cases, Right table keys have a short life cycle we called it as session_id, the session_id will be dead after 2 min. If every session_id in this operator only clean leftState and rightState, the nextLeftIndex state and registeredTimer state of those session_ids will be stored forever. After run for a day(20 million session_id), all session_id's nextLeftIndex and  registeredTimer state data will cause the job crash.
I will add the test for this case in TemporalRowTimeJoinOperatorTest @leonardBang

Reply:
Thanks for you clarification @zicat , make sense to me. 
BTW, the PR title and description need more clear.

Reply:
I have added the test in TemporalRowTimeJoinOperatorTest, please help to review it @leonardBang .
I will modify the title and description.

Reply:
> Thanks for you clarification @zicat , make sense to me.
> BTW, the PR title and description need more clear.

Done, Please help to review again.

Reply:
Please let me know if there is anything I should change, thx @leonardBang

Reply:
left minor comments